### PR TITLE
Created SkillsRatingsComponent 

### PR DESCRIPTION
**Pull request by kwiedenhaupt**
1. Created the SkillsRatingsComponent 
2. Fetched user data to display the skills if there was data
3. Created a query to FetchAllSkills in case the user has no skills.
4. Massaged the data so that the current upsert query could be used in conjunction with the new data from FetchAllSkills to create a new entry in the intro_course_skills_users table if there was no data in the table for that user.
5. Saved the data in a local state
6. used the UseLazyQuery on the FetchAllAssignments so that it only runs if the user has no data in the table.
7. Modified profileV2 with the new SkillsRatingsComponent

This is how skills are currently displaying on the profileV2 page:
Screen Shot 2022-12-21 at 8 59 25 AM

### CODE REVIEW

File: components/coding/SkillRatingsComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  const {} = useQuery<FetchUserSkillsRatings>(FETCH_USER_SKILLS_RATINGS, {
+    variables: {
+      userId: user.uid,
+    },
+    onCompleted: (data) => {
+      const currentSkillsFiltered = data.intro_course_skills_user.filter(
+        (it) => it.intro_course_skill.id
+      );
+      console.log("CSF", currentSkillsFiltered);
+      const currentSkillsMapped = data.intro_course_skills_user.map(
```
Comment (iamnotreddy) [inline components/coding/SkillRatingsComponent.tsx]: naming conventions are generally confusing, can we do something like: allSkills, currentlyRatedSkills, missingSkills

**Review by iamnotreddy — CHANGES_REQUESTED** [review]
I think you did a lot of good stuff here!

Here's what I'm thinking to solve for the "update" issue inside the onComplete of FETCH_USER_SKILLS_RATINGS:
- if all the skills match all the skills with user ratings, there are no missing skills; setSkillRatings just like original
- if there are no user skill ratings, initialize the skill ratings in an object and set them in the skillRatings
- if there are some missing skill ratings, initialize the missing ones in one object, then combine that object with the existing skill ratings, and setSkillRatings to that new combined object

This way you shouldn't have to worry about refreshing. This also means that we won't always save the newly initialized skill ratings unless the user presses save, but I think that's fine because we're checking whether there are any missing skills each time this component loads

File: components/coding/SkillRatingsComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import { useQuery, useMutation, useLazyQuery } from "@apollo/client";
+import { createHook } from "async_hooks";
```
[the remark this answers is not in the document]
Comment (kwiedenhaupt) [reply components/coding/SkillRatingsComponent.tsx]: removed

File: components/coding/SkillRatingsComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  skillRatingsSelector,
+  setSkillRatings,
+  SkillRatingsRow,
+  updateSkillRatings,
+} from "../../redux/skillRatingsSlice";
+import SkillSection from "../skillRatings/SkillSection";
+import { Button } from "../ui/Button";
+import { FetchAllSkills, FETCH_ALL_SKILLS } from "./FetchAllSkills";
+
+const initializeSkillRating = (skillIds: string[], userId: string) => {
```
[the remark this answers is not in the document]
Comment (kwiedenhaupt) [reply components/coding/SkillRatingsComponent.tsx]: done

File: components/coding/SkillRatingsComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    FETCH_ALL_SKILLS,
+    {
+      onCompleted: (data) => {
+        const skillIds = data.intro_course_skills.map((it) => it.id);
+
+        setSkillIds(skillIds);
+      },
+    }
+  );
+  const [skillIds, setSkillIds] = useState<string[]>([]);
```
[the remark this answers is not in the document]
Comment (kwiedenhaupt) [reply components/coding/SkillRatingsComponent.tsx]: moved it.

File: components/coding/SkillRatingsComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  const dispatch = useDispatch();
+  const { user } = useAuth();
+  const { skillRatings } = useSelector(skillRatingsSelector);
+
+  const {} = useQuery<FetchUserSkillsRatings>(FETCH_USER_SKILLS_RATINGS, {
+    variables: {
+      userId: user.uid,
+    },
+    onCompleted: (data) => {
+      if (data.intro_course_skills_user !== null) {
```
[the remark this answers is not in the document]
Comment (kwiedenhaupt) [reply components/coding/SkillRatingsComponent.tsx]: changed

File: components/coding/SkillRatingsComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  const [skillIds, setSkillIds] = useState<string[]>([]);
+  const [currentSkills, setCurrentSkills] = useState<string[]>([]);
+
+  const {} = useQuery<FetchUserSkillsRatings>(FETCH_USER_SKILLS_RATINGS, {
+    variables: {
+      userId: user.uid,
+    },
+    onCompleted: (data) => {
+      const currentSkillsFiltered = data.intro_course_skills_user.filter(
+        (it) => it.intro_course_skill.id
```
[the remark this answers is not in the document]
Comment (kwiedenhaupt) [reply components/coding/SkillRatingsComponent.tsx:47]: removed.

File: components/coding/SkillRatingsComponent.tsx (hunk 1 above)
Comment (kwiedenhaupt) [reply components/coding/SkillRatingsComponent.tsx]: changed to as suggested.

**Comment by kwiedenhaupt** [comment]
Took a different approach focused on getting all the relevant information needed in one query and then transforming data to the SkillRatingsRow[].  Seems to work!
Screen Shot 2022-12-29 at 4 37 34 PM

File: components/coding/SkillRatingsComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
+import { useQuery, useMutation } from "@apollo/client";
+import React from "react";
```
Comment (iamnotreddy) [inline components/coding/SkillRatingsComponent.tsx:2]: don't think you actually need this in here

**Review by iamnotreddy — APPROVED** [review]
great work! 

looks like you have a merge conflict you'll need to fix, also check my comment on the import

**Pull request merged**